drivers/SmartThings/zigbee-thermostat/src/leviton/init.lua

```diff
@@ -89,12 +90,12 @@ end
 
 local function set_cooling_setpoint(driver, device, command)
   device:emit_event(capabilities.thermostatCoolingSetpoint.coolingSetpoint({value = command.args.setpoint*1.0, unit = "C"}))
-  device:send_to_component(command.component, Thermostat.attributes.OccupiedCoolingSetpoint:write(device, command.args.setpoint*100))
+  device:send_to_component(command.component, Thermostat.attributes.OccupiedCoolingSetpoint:write(device, utils.round(command.args.setpoint*100)))
 end
 
 local function set_heating_setpoint(driver, device, command)
   device:emit_event(capabilities.thermostatHeatingSetpoint.heatingSetpoint({value = command.args.setpoint*1.0, unit = "C"}))
-  device:send_to_component(command.component, Thermostat.attributes.OccupiedHeatingSetpoint:write(device, command.args.setpoint*100))
+  device:send_to_component(command.component, Thermostat.attributes.OccupiedHeatingSetpoint:write(device, utils.round(command.args.setpoint*100)))
```

Comment:
Are these events being rejected because of this? I assume thats the reason for the change, but if not then it seems like we are reducing the precision for users, which I know has been frustrating in the past.

Reply:
1. Yes the events were being rejected, because this `write` expects an Int, and we were sending a float
2. There's no loss of precision here, since zigbee expects tenths of a degree here (we're rounding after multiplying by 100).